src/scss/_utils.scss

```
@@ -51,8 +51,17 @@
   margin-top: 0.25rem;
 }
 
+// Limit the width of tooltip to 20rem.
+// 25rem was chosen as the breakpoint, because 80vw equals 20rem at 25rem.
 .p-tooltip--constrain-width .p-tooltip__message {
-  max-width: 20rem;
+  @media (width <= 25rem) {
+    width: 80vw;
+  }
+
+  @media (width > 25rem) {
+    width: 20rem;
+  }
+
   white-space: pre-wrap;
```

I'm not sure why you were getting tooltips with 0 width (maybe we can talk about this later), but I think we can improve this slightly by using the predefined breakpoints. In this case we probably want to change the tooltip at the same breakpoint that the panel goes full width.

I also think we probably want to handle this differently to the `p-tooltip--constrain-width` tooltip so that those tooltips don't get a fixed width.

The issue with using a fixed with is that the tooltips then don't reduce the width to smaller content. In MAAS I think we made [this util](https://github.com/canonical/maas-ui/blob/main/src/app/utils/breakLines.ts) to set a max width on the lines before they were inserted into the tooltip.

Here's how we could implement this using the existing breakpoints:

```suggestion
.p-tooltip--constrain-width .p-tooltip__message {
  max-width: 20rem;
  white-space: pre-wrap;

  @media (max-width: $breakpoint-x-small) {
    max-width: 80vw;
  }
}

.p-tooltip--fixed-width .p-tooltip__message {
  white-space: pre-wrap;
  width: 20rem;

  @media (max-width: $breakpoint-x-small) {
    width: 80vw;
  }
}
```